core/src/main/java/io/grpc/Server.java

```diff
@@ -38,4 +41,57 @@
  * application code or interceptors.
  */
 @ThreadSafe
-public abstract class Server {}
+public abstract class Server {
```

Comment:
On the client side we need the Channel/ManagedChannel separation because we have different use cases in mind, most of which don't need to (nor should they be able to) modify channel lifecycle.  For servers, we don't have those use cases and the server is only used by the main app that controls server lifecycle. Again, it doesn't seem quite symmetrical and `Server` seems like a fine name to me.